open_optical_gating/cli/optical_gater_server.py

```diff
@@ -105,30 +107,41 @@ def initialise_internal_parameters(self):
         self.initial_process_time_s = time.time()
 
         # Defines variables and objects used for plotting
-        self.timestamp = []
-        self.phase = []
-        self.processing_rate_fps = []
-        self.trigger_times = []
-        self.predicted_trigger_time_s = []
+        self.processing_rate_fps = []  # TODO move into pixelarray object?
+        self.sent_trigger_times = []
+        self.predicted_trigger_time_s = []  # TODO move into pixelarray object?
 
         # Flag for interrupting the program at key points
         # E.g. when user-input is needed
         # It is assumed that the user/app controls what this interaction is
         self.stop = False
 
-    def analyze(self, pixelArray):
+    def analyze_pixelarray(self, pixelArray):
         """ Method to analyse each frame as they are captured by the camera.
             The documentation explains that this must be fast, since it is running within the encoder's callback,
             and so must return before the next frame is produced.
             Essentially this method just calls through to another appropriate method, based on the current value of the state attribute."""
-        logger.debug("Analysing frame.")
+        logger.debug(
+            "Analysing frame.; Current time: {0}s", pixelArray.metadata["timestamp"],
+        )
 
         # For logging processing time
         time_init = time.time()
 
+        # TODO: These lines need to be moved into the eventual
```

Comment:
Are you intending to do something about this now (i.e. have you just forgotten to finish this) or is your intention to leave this for later? On the face of it, it looks important for the Pi!

Reply:
I only made these a comment recently. I would like to leave them there until I work out how the pi_optical_gater is actually going work.